staging/src/k8s.io/client-go/tools/cache/controller.go

```diff
@@ -704,6 +709,9 @@ func processDeltasInBatch(
 //   - clientState is the store you want to populate
 //   - options contain the options to configure the controller
 func newInformer(clientState Store, options InformerOptions) Controller {
+	if options.Name == "" {
+		options.Name = naming.GetNameFromCallsite(internalPackages...)
```

Comment:
What if the name is duplicated? IIUC the metric will be flapping in this case between all known values

based on the example, it looks like it may be duplicated quite often. Also the name seems to be quite duplicate of item_type.

```
fifo_queued_items{item_type="v1.CSIDriver",name="k8s.io/client-go/informers/storage/v1/csidriver.go:58"} 0
```


Maybe we can migrate the name to some pattern where we use unique numbers? It seems to have similar information for metric users and addresses the name duplication concern.